src/transpiler/resolve_tb_fusion.cc

```diff
@@ -0,0 +1,87 @@
+#include "mirage/transpiler/transpiler.h"
+
+#include "mirage/transpiler/utils.h"
+#include "mirage/type.h"
+
+namespace mirage {
+namespace transpiler {
+
+// Decide when and how to fuse operators in every threadblock level op
+void Transpiler::resolve_tb_fusion() {
+  for (kn::KNOperator *const op : g->operators) {
+    if (op->op_type != type::KN_CUSTOMIZED_OP) {
+      continue;
+    }
+    kn::KNCustomizedOp *const custom_op =
+        dynamic_cast<kn::KNCustomizedOp *>(op);
+    tb::Graph const &tb_graph = custom_op->bgraph;
+
+    std::unordered_map<sguid_t, int> num_consumers;
+    // Initialization
+    for (tb::TBOperator *const op : tb_graph.operators) {
+      is_fused_with_prev[op] = false;
+      is_fused_with_next[op] = false;
+      for (tb::STensor const &stensor :
+           Combine(op->input_tensors, op->output_tensors)) {
+        num_consumers[stensor.guid] = 0;
+      }
+    }
+    // Count the number of consumers for each tensor
+    for (tb::TBOperator *const op : tb_graph.operators) {
+      for (tb::STensor const &input : op->input_tensors) {
+        num_consumers[input.guid] += 1;
+      }
+    }
+    // Currently we only fuse elementwise unary operators (exp / accum) with
+    // the previous operator, when the output of the previous operator has only
+    // one consumer
+    for (tb::TBOperator *const op : tb_graph.operators) {
+      if ((op->op_type == type::TB_EXP_OP ||
+           op->op_type == type::TB_FORLOOP_ACCUM_NO_RED_OP)) {
+        tb::STensor const &input0 = op->input_tensors.at(0);
+        tb::TBOperator *prev_op = input0.owner_op;
+        // Don't fuse with an input op with forloop_dim = -1
+        if (prev_op->op_type == type::TB_INPUT_OP &&
+            dynamic_cast<tb::TBInputOp *>(prev_op)->forloop_dim == -1) {
+          continue;
+        }
```

Comment:
Do we still need this part in the case where there is already one in line 54

Reply:
Yes, this is indeed unnecessary, but I leave the code here as a reminder: we are never going to fuse with an input op with forloop_dim = -1